src/_guides/language/effective-dart/design.md

```diff
@@ -1005,90 +1089,254 @@ var names = people.map((Person person) {
-  var value = map[name];
-  if (value != null) return value;
-  return defaultValue;
+void handleError([!void Function()!] operation, [!Function!] errorHandler) {
+  try {
+    operation();
+  } catch (err, stack) {
+    if (errorHandler is [!Function(Object)!]) {
+      errorHandler(err);
+    } else if (errorHandler is [!Function(Object, Object)!]) {
+      errorHandler(err, stack);
+    }
+  }
 }
 {% endprettify %}
 
 
-### AVOID annotating with `Function`.
+### DON'T specify a return type for a setter.
 
-The `Function` type is barely more precise than using no annotation at all. If
-you're bothering to annotate, it's better to use a precise function type that
-describes the signature and return type of the function.
+Setters always return `void` in Dart. Writing the word is pointless.
 
-If you are annotating a field, this does mean you have to create a typedef, but
-that's usually worth doing.
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (avoid_return_types_on_setters)"?>
+{% prettify dart %}
+void set foo(Foo value) { ... }
+{% endprettify %}
 
 {:.good-style}
-<?code-excerpt "misc/lib/effective_dart/design_good.dart (avoid-Function)"?>
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (avoid_return_types_on_setters)"?>
 {% prettify dart %}
-bool isValidString(String value, bool predicate(String string)) => ...
+set foo(Foo value) { ... }
 {% endprettify %}
 
+
+### DON'T use the legacy typedef syntax.
+
+Dart has two notations for defining a named typedef for a function type. The
+original syntax looks like:
+
 {:.bad-style}
-<?code-excerpt "misc/lib/effective_dart/design_bad.dart (avoid-Function)"?>
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (old-typedef)"?>
+{% prettify dart %}
+typedef int Comparison<T>(T a, T b);
+{% endprettify %}
+
+That syntax has a couple of problems:
+
+*   There is no way to assign a name to a *generic* function type. In the above
+    example, the typedef itself is generic. If you reference `Comparison` in
+    your code, without a type argument, you implicitly get the function type
+    `int Function(dynamic, dynamic)`, *not* `int Function<T>(T, T)`. This
+    doesn't come up in practice often, but it matters in certain corner cases.
+
+*   A single identifier in a parameter is interpreted as the parameter's *name*,
+    not its *type*. Given:
+
+    {:.bad-style}
+    <?code-excerpt "misc/lib/effective_dart/design_bad.dart (typedef-param)"?>
+    {% prettify dart %}
+    typedef bool TestNumber(num);
+    {% endprettify %}
+
+    Most users expect this to be a function type that takes a `num` and returns
+    `bool`. It is actually a function type that takes *any* object (`dynamic`)
+    and returns `bool`. The parameter's *name* (which isn't used for anything
+    except documentation in the typedef) is "num". This has been a
+    long-standing source of errors in Dart.
+
+The new syntax looks like:
```

Comment:
like -> like this

(The colon should be preceded by a sentence: https://developers.google.com/style/colons)

Reply:
Done.